### DISCUSSION TITLE
Can I create rule for complex

### DISCUSSION THREAD

**Discussion by szymic1** [zeroSteiner/rule-engine#27] [Q&A] [answered]
Hi,

I have such one element:
`ledger = [
    {
    "how_many_km": {
            "type": "integer",
            "value": "100000",
            "visibility": "public",
            "entry_created_ts": 213423423424.123,
            "entry_updated_ts": 213423423424.123,
            "value_changed_ts": 213423423424.123
        },
    "colour": {
        "type": "string",
        "value": "red",
        "visibility": "public",
        "entry_created_ts": 213423423424.123,
        "entry_updated_ts": 213423423424.123,
        "value_changed_ts": 213423423424.123
    },
    "type_of_engine": {
        "type": "string",
        "value": "diesel",
        "visibility": "public",
        "entry_created_ts": 213423423424.123,
        "entry_updated_ts": 213423423424.123,
        "value_changed_ts": 213423423424.123
    }
  }
]`

and I would like to check if such item satisfy  condition:
how_many_km.value=="10000" and colour.value=='red' or  type_of_engine.value=="gas"

Is it possible to use such complex object in rule? If yes where do I find example how to create such rule?

Regards 
Michał Szymański

**Comment by szymic1** [comment, marked as answer]
Ok, I've found solution:

rule = rule_engine.Rule(
  'how_many_km.value == "100000" and colour.value=="red" or type_of_engine.value=="gas"'
)

**Reply by zeroSteiner** [reply]
Yup that looks correct to me.
